src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -152,68 +99,7 @@ export class SavedObjectsClient {
    * @returns {promise} - [{ id, type, version, attributes, error: { message } }]
    */
   async bulkCreate(objects, options = {}) {
-    const {
-      overwrite = false
-    } = options;
-    const time = this._getCurrentTime();
-    const objectToBulkRequest = (object) => {
-      const method = object.id && !overwrite ? 'create' : 'index';
-
-      return [
-        {
-          [method]: {
-            _id: this._generateEsId(object.type, object.id),
-            _type: this._type,
-          }
-        },
-        {
-          type: object.type,
-          updated_at: time,
-          [object.type]: object.attributes
-        }
-      ];
-    };
-
-    const { items } = await this._writeToCluster('bulk', {
-      index: this._index,
-      refresh: 'wait_for',
-      body: objects.reduce((acc, object) => ([
-        ...acc,
-        ...objectToBulkRequest(object)
-      ]), []),
-    });
-
-    return items.map((response, i) => {
-      const {
-        error,
-        _id: responseId,
-        _version: version,
-      } = Object.values(response)[0];
-
-      const {
-        id = responseId,
-        type,
-        attributes,
-      } = objects[i];
-
-      if (error) {
-        return {
-          id,
-          type,
-          error: {
-            message: error.reason || JSON.stringify(error)
-          }
-        };
-      }
-
-      return {
-        id,
-        type,
-        updated_at: time,
-        version,
-        attributes
-      };
-    });
+    return await this._repository.bulkCreate(objects, options);
```

Comment:
nit: You don't need the `await` on these return statements. When you return a promise in an async function, it is chained. Or that original promise is what is returned. I can't really remember, but the net effect is the same in this case.

Reply:
This is a lint issue that we can solve with https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md

Reply:
Yeah... I've been torn on this before, but added it here to hope it'd help readability. I wanted the methods themselves to continue to use the `async` keyword (because it makes it obvious that it's returning a Promise) but it felt weird to just do a return `this._repository.bulkCreate(...)` because then it's non-obvious where the asynchronicity comes in.

Reply:
Unfortunately, while it increases readability when using `await`, using `return await` is actually a performance degradation, and worse, people reading this will assume it's a pattern they should use.

Reply:
But again--I think the lint rule will make it so we don't need to worry about it.

Reply:
Yeah... not fond of that rule, I'm interested in what type of performance impact this actually has.

Reply:
I decided to stop protesting, and I went ahead and removed the awaits.

Reply:
I thoroughly disagree with the premise of https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md.

I think the `return await` is more explicit and therefor more readable and understandable.

And finally, there are subtle differences between an async function that returns a promise and one that `return awaits` which I think most people are not familiar with and can lead to subtle bugs https://jakearchibald.com/2017/await-vs-return-vs-return-await/.

If it was possible to enforce that promises were always awaited before being returned I would absolutely be fighting to add that to our eslint config.

Reply:
This sort of thing can be dealt with in a separate issue, which is why I specifically called this out as a nitpick. As a general rule of thumb, if a nitpick spawns a debate about linting rules, it should probably prompt the author to just ignore it for now.